src/context/SWR.tsx

```diff
@@ -18,27 +16,32 @@ export const DEFAULT_POLLING = 2500;
 
 const SwrConfigProvider = ({ children }: BaseComponentProps) => {
     const supabaseClient = useClient();
-    const navigate = useNavigate();
+    const intl = useIntl();
+    const { enqueueSnackbar } = useSnackbar();
 
     const cache = () => {
         return new LRU({
             max: 500,
         });
     };
 
-    const errorHandler = (error: any) => {
+    const errorHandler = async (error: any) => {
         // Handle JWT tokens expiring
         if (error.message === ERROR_MESSAGES.jwtExpired) {
-            supabaseClient.auth
+            await supabaseClient.auth
                 .signOut()
                 .then(() => {
-                    navigate(
-                        getPathWithParam(
-                            logoutRoutes.path,
-                            logoutRoutes.params.reason,
-                            LogoutReasons.JWT
-                        ),
-                        { replace: true }
+                    enqueueSnackbar(
+                        intl.formatMessage({
+                            id: 'login.jwtExpired',
+                        }),
+                        {
+                            anchorOrigin: {
+                                vertical: 'top',
+                                horizontal: 'center',
+                            },
+                            variant: 'error',
+                        }
                     );
```

Comment:
Just making the thing that is forcing the log out responsible for the notification. Makes a lot of stuff simpler. Also, with notistack we can have references to what opened it and put in custom actions so this empowers some interesting possibilities.